extensions/amp-social-share/1.0/social-share.js

```diff
@@ -24,33 +24,36 @@ import {getSocialConfig} from './social-share-config';
 import {openWindowDialog} from '../../../src/dom';
 import {useResourcesNotify} from '../../../src/preact/utils';
 
+const NAME = 'SocialShare';
 const DEFAULT_WIDTH = 60;
 const DEFAULT_HEIGHT = 44;
 const DEFAULT_TARGET = '_blank';
-const NAME = 'SocialShare';
+const WINDOW_FEATURES = 'resizable,scrollbars,width=640,height=480';
 
 /**
- * @param {!JsonObject} props
- *  type: !string,
- *  endpoint: ?string,
- *  params: ?JsonObject,
- *  target: ?string,
- *  width: ?string,
- *  height: ?string,
- *  color: ?string,
- *  background: ?string,
- *  tabIndex: ?string,
- *  style: ?string,
+ * @param {!SocialShareProps} props
  * @return {PreactDef.Renderable}
  */
-export function SocialShare(props) {
+export function SocialShare({
+  type,
+  endpoint,
+  params,
+  target,
+  width,
+  height,
+  color,
+  background,
+  tabIndex,
+  style,
+  children,
```

Comment:
Let's also propagate `...rest`

Reply:
Thanks, I see this is a common pattern for our other components.  For this specific instance, I'm not exactly sure where we would want to propagate it too.  For the most part, we don't really want the icon to change that much except for colors that we allow, nor do we want to change the wrapper `div` that much.  If there's something I'm missing, could you elaborate?

Reply:
There could be a number of attributes we want to support generically, i.e. `aria-label`, `aria-disabled`, `disabled` -- if this is a button I assume we want to respect if it's not clickable? I don't think we should try to support these exhaustively, hence `rest`. If there are ones we explicitly do not want to support then that is worth discussing too.

Reply:
Sounds good, added!